### PR TITLE
Fix providers & improve logging

### PR DESCRIPTION
**Pull request by martinvy**
- Fixer now requires access token. I created an account and added credentials to Jenkins & Kube.
- Google blocks us because we do automatic requests. I tried to lower the traffic.
- Yahoo service doesn't work anymore, but I didn't find any new way how to get the rates because the new webpage didn't really work for me when I tried so I am leaving fixing of Yahoo for later.

- I improved some logs and I plan to set up alert with notification to Slack.

@business-factory/pythonistas please review & merge.

### CODE REVIEW

File: gold_digger/data_providers/fixer.py [hunk 1]
```diff
@@ -7,17 +7,29 @@
...
+    Base currency is in EUR and cannot be changed in free subscription.
+    We have to convert exchange rates to base currency (USD) before returning the rates from the provider.
+    """
+    BASE_URL = "http://data.fixer.io/api/{date}?access_key=%s"
     name = "fixer.io"
 
+    def __init__(self, access_key, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if access_key:
+            self._url = self.BASE_URL % access_key
```
Comment (harpener) [inline gold_digger/data_providers/fixer.py]: @martinvy It's just a detail, but you don't really need formatting, if you are appending text at the end. :-)

File: gold_digger/data_providers/fixer.py [hunk 2]
```diff
@@ -7,17 +7,29 @@
...
-    BASE_URL = "https://api.fixer.io/{date}"
+    """
+    Base currency is in EUR and cannot be changed in free subscription.
+    We have to convert exchange rates to base currency (USD) before returning the rates from the provider.
+    """
+    BASE_URL = "http://data.fixer.io/api/{date}?access_key=%s"
     name = "fixer.io"
 
+    def __init__(self, access_key, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (harpener) [inline gold_digger/data_providers/fixer.py]: @martinvy You could be explicit about what parameters are passed here :-)

Comment (martinvy) [reply gold_digger/data_providers/fixer.py]: That's C&P from CurrencyLayer.

File: gold_digger/data_providers/fixer.py (hunk 1 above)
Comment (martinvy) [reply gold_digger/data_providers/fixer.py]: Yes, there are more ways how to format text :D

File: gold_digger/data_providers/fixer.py [hunk 3]
```diff
@@ -91,18 +108,30 @@ def _get_by_date(self, date_of_exchange, currency):
...
 
         return None
 
+    def _conversion_to_base_currency(self, base_currency_rate, currency_rate):
+        """
+        :type base_currency_rate: decimal.Decimal
+        :type currency_rate: decimal.Decimal
+        :rtype: decimal.Decimal
+        """
+        conversion = 1 / base_currency_rate
```
Comment (harpener) [inline gold_digger/data_providers/fixer.py]: @martinvy We aren't really filtering zero values here, but I guess they make no sense. :-)

**Pull request merged**